Mathlib/GroupTheory/Perm/Centralizer.lean

```diff
@@ -0,0 +1,893 @@
+        ((τ : Perm g.cycleFactorsFinset) d : Perm α) :=
+        cycleFactorsFinset_pairwise_disjoint g ((τ : Perm g.cycleFactorsFinset) c).prop
+          ((τ : Perm g.cycleFactorsFinset) d).prop (by
+          intro h'; apply h
+          simpa only [Subtype.coe_inj, EmbeddingLike.apply_eq_iff_eq] using h')
+      rw [disjoint_iff_disjoint_support, Finset.disjoint_right] at H H'
+      simp only [H hd, H' (mem_support_self a _)]
+
+theorem newK_apply_mem_fixedPoints_iff {x : α} :
+    newK a τ x ∈ Function.fixedPoints g ↔ x ∈ Function.fixedPoints g := by
+  rcases mem_fixedPoints_or_exists_zpow_eq a x with (hx | ⟨c, hc, m, hm⟩)
+  · simp only [hx, newK_apply_of_mem_fixedPoints a τ hx]
+  · rw [newK_apply_of_cycleOf_mem a τ hc hm, ← hm]
+    simp only [Function.mem_fixedPoints_iff, ← not_mem_support]
+    simp only [zpow_apply_mem_support, not_iff_not]
+    simp only [Finset.coe_mem,
+      mem_cycleFactorsFinset_support_le _ (mem_support_self a _)]
+
+theorem newK_commute_zpow_apply (x : α) (j : ℤ) :
+    newK a τ ((g ^ j) x) = (g ^ j) (newK a τ x) := by
+  rcases mem_fixedPoints_or_exists_zpow_eq a x with (hx | hx)
+  · rw [newK_apply_of_mem_fixedPoints a τ hx, newK_apply_of_mem_fixedPoints]
+    rw [Function.mem_fixedPoints_iff]
+    simp only [← mul_apply, ← zpow_one_add, add_comm]
+    conv_rhs => rw [← hx, ← mul_apply, ← zpow_add_one]
+  · obtain ⟨c, hc, m, hm⟩ := hx
+    have hm' : (g ^ (j + m)) (a c) = (g ^ j) x := by rw [zpow_add, mul_apply, hm]
+    rw [newK_apply_of_cycleOf_mem a τ hc hm, newK_apply_of_cycleOf_mem a τ _ hm',
+      ← mul_apply, ← zpow_add]
+    exact zpow_apply_mem_support_of_mem_cycleFactorsFinset_iff.mpr hc
+
+theorem newK_mul (σ τ : range_toPermHom' g) (x) :
+    newK a (σ * τ) x = (newK a σ) (newK a τ x) := by
+  rcases mem_fixedPoints_or_exists_zpow_eq a x with (hx | ⟨c, hc, m, hm⟩)
+  · simp only [newK_apply_of_mem_fixedPoints a _ hx]
+  · simp only [newK_apply_of_cycleOf_mem a _ hc hm]
+    rw [newK_apply_of_cycleOf_mem a _ _ rfl]
+    · rfl
+    · rw [zpow_apply_mem_support_of_mem_cycleFactorsFinset_iff]
+      apply mem_support_self
+
+theorem newK_one (x : α) : (newK a 1) x = x := by
+  rcases mem_fixedPoints_or_exists_zpow_eq a x with (hx | ⟨c, hc, m, hm⟩)
+  · rw [newK_apply_of_mem_fixedPoints a _ hx]
+  · rw [newK_apply_of_cycleOf_mem a _ hc hm, OneMemClass.coe_one, coe_one, id_eq, hm]
+
+/-- Given `a : g.Basis` and a permutation of `g.cycleFactorsFinset` that
+  preserve the lengths of the cycles, a permutation of `α` that
+  moves the `Basis` and commutes with `g` -/
+noncomputable def ofPermHom : range_toPermHom' g →* Perm α where
+  toFun τ := {
+    toFun := newK a τ
+    invFun := newK a τ⁻¹
+    left_inv := fun x ↦ by rw [← newK_mul, inv_mul_cancel, newK_one]
+    right_inv := fun x ↦ by rw [← newK_mul, mul_inv_cancel, newK_one] }
+  map_one' := ext fun x ↦ newK_one a x
+  map_mul' := fun σ τ ↦ ext fun x ↦ by simp [mul_apply, newK_mul a σ τ x]
+
+theorem ofPermHom_support :
```

Comment:
It seems like this lemma and the following lemma aren't used?

Reply:
indeed